File: game-server/src/com/aionemu/gameserver/network/aion/capture/ServerPacketCaptureObserver.java
package com.aionemu.gameserver.network.aion.capture;

import java.nio.ByteBuffer;

import com.aionemu.gameserver.network.aion.AionConnection;
import com.aionemu.gameserver.network.aion.AionServerPacket;

/**
 * Java parity capture hook for optional server-packet runtime artifacts.
 */
public interface ServerPacketCaptureObserver {

	boolean isEnabled();

	void onPacketSerialized(AionConnection con, AionServerPacket packet, ByteBuffer clearFrame);
}

File: dotnetConversion/src/Aion.GameServer/Network/Aion/Capture/ServerPacketCaptureObserver.cs
using Aion.Commons.Nio;
using Aion.GameServer.Network.Aion;

namespace Aion.GameServer.Network.Aion.Capture;

/// <summary>Java parity: network/aion/capture/ServerPacketCaptureObserver. Observes serialized server packets.</summary>
public interface ServerPacketCaptureObserver
{
    bool IsEnabled();
    void OnPacketSerialized(AionConnection con, AionServerPacket packet, ByteBuffer clearFrame);
}
